streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -198,12 +198,19 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
         }
 
         if (!taskCloseExceptions.isEmpty()) {
+            for (final Map.Entry<TaskId, RuntimeException> entry : taskCloseExceptions.entrySet()) {
+                if (!(entry.getValue() instanceof TaskMigratedException)) {
```

Comment:
I think we should use `instanceof KafkaException` as well here:  if it is any inheritance of KafkaException, we should not wrap it.

Reply:
Sounds good